lib/project_types/extension/commands/create.rb

```diff
@@ -22,19 +22,12 @@ def call(args, _)
             @ctx.abort(message_for_extension["create.errors.directory_exists", form.directory_name])
           end
 
-          return use_new_create_flow(form, message_for_extension) if supports_development_server?(form.type)
-
-          if form.type.create(form.directory_name, @ctx, getting_started: options.flags[:getting_started])
-            ExtensionProject.write_cli_file(context: @ctx, type: form.type.identifier)
-            ExtensionProject.write_env_file(
-              context: @ctx,
-              title: form.name,
-              api_key: form.app.api_key,
-              api_secret: form.app.secret
-            )
-
-            @ctx.puts(message_for_extension["create.ready_to_start", form.directory_name, form.name])
-            @ctx.puts(message_for_extension["create.learn_more", form.type.name])
+          if supports_development_server?(form.type)
```

Comment:
I was hoping to use `Shopify::Result` to chain this flow, like so:
```ruby
          ShopifyCli::Result
            .call(&method(:supports_development_server?).curry[form.type])
            .then(&method(:create_extension).curry[form])
            .then(&method(:write_env_file).curry[form])
            .then(&method(:notify_user).curry[form, message_for_extension])
            .unwrap do |_error|
              @ctx.puts(message_for_extension["create.try_again"])
            end
```
but I am getting an error when I run `supports_development_server`: `/Users/cecycorrea/src/github.com/Shopify/shopify-cli/lib/project_types/extension/commands/create.rb:26:in `block in call': wrong argument type ShopifyCli::Result::Success (expected Proc) (TypeError)`. Not sure why it is expecting a `Proc` there, as this is a pattern I've used before.

If it's more trouble than it's worth at the moment to use the `Result` object, we can keep this as is though.